lang/cs/Org.Apache.REEF.Client/YARN/IJobSubmissionDirectoryProvider.cs

```diff
@@ -29,6 +29,6 @@ public interface IJobSubmissionDirectoryProvider
         /// Returns path to job submission directory in DFS.
         /// </summary>
         /// <returns></returns>
-        string GetJobSubmissionRemoteDirectory();
+        string GetJobSubmissionRemoteDirectory(string applicationId);
```

Comment:
The client (`YarnREEFDotNetClient`) during `Submit` (or similar call in the future).